### ISSUE TITLE
It is possible to use sphinx.ext.autosectionlabel with sphinx.ext.intersphinx?

### ISSUE THREAD

**Issue by jputrino** [sphinx-doc/sphinx#2579]
I'd like to use the autosectionlabel and intersphinx extensions to easily cross-reference related but separate documentation sets. Currently, I'm only able to cross-reference via intersphinx if I've explicitly created the section header. Do you plan to support using autosectionlabel with intersphinx?

Example: 
In f5-openstack-agent/docs/index.rst, I manually added the `.. _home:` section label. When I use 

```
:ref:`F5 Agent <agent:home>` 
```

to cross-reference from f5-openstack-lbaasv2-driver via intersphinx, it works fine. 

In f5-openstack-agent/docs/.../topic_install-the-f5-agent.rst, I have not explicitly defined a section header.  The first heading on the page is 'Install the F5 OpenStack Agent'. When I use 

```
:ref:`agent:Install the F5 OpenStack Agent`
```

to cross-reference the page from f5-openstack-lbaasv2-driver, I get the following warning: 

```
WARNING: undefined label: agent:install the f5 openstack agent (if the link has no caption the label must precede a section header)
```

thanks!

**Comment by tk0miya** [comment]
I can refer the section by its name locally using `sphinx.ext.autosectionlabel` and `sphinx.ext.intersphinx`.

Did you enable `sphinx.ext.autosectionlabel` on `f5-openstack-lbaasv2-driver` ?
The extension creates labels to each section automatically as its name suggests. So refered project has to load it.

**Comment by jputrino** [comment]
Hi @tk0miya. I finally have an example for you. I have `sphinx.ext.autosectionlabel` enabled in `f5-openstack-lbaasv2-driver` and `f5-openstack-heat`, and I have both set up in intersphinx mapping. 

The following are in conf.py in both repos:

```
extensions = [
    'sphinx.ext.autodoc',
    'sphinx.ext.doctest',
    'sphinx.ext.intersphinx',
    'sphinx.ext.todo',
    'sphinx.ext.coverage',
    'sphinx.ext.viewcode',
    'sphinx.ext.autosectionlabel',
]
```

```
intersphinx_mapping = {'heat': (
     'http://f5-openstack-heat.readthedocs.io/en/latest', None),
     'heatplugins': (
     'http://f5-openstack-heat-plugins.readthedocs.io/en/liberty', None),
     'lbaasv1': (
     'http://f5-openstack-lbaasv1.readthedocs.io/en/liberty/', None),
     'agent': (
     'http://f5-openstack-agent.readthedocs.io/en/liberty/', None),
     'f5sdk': (
     'http://f5-sdk.readthedocs.io/en/latest/', None),
     'docs': (
     'http://f5-openstack-docs.readthedocs.io/en/liberty/', None),
 }
```

When I run `sphinx-build` locally, I get the `undefined label` warning for these refs. I've done it both with the ® in the ref and without it, with no difference. I'll also try removing the ® from the title in the doc I'm trying to ref.

![2016-06-21_09-31-35](https://cloud.githubusercontent.com/assets/12143793/16235799/53ea8c28-3793-11e6-8c5b-ac25563e477f.jpg)

**Comment by tk0miya** [comment]
Thank you. I could reproduce the problem. I'll take a look later.

In my short investigaton:
I can refer "Overview" section of f5-openstack-heat doc using autosectionlabel and intersphinx
But I can't refer the "f5-openstack-heat" section.

```
$ sphinx-quickstart test
$ cd test
$ cat <<EOS >> conf.py
extensions.append('sphinx.ext.intersphinx')
intersphinx_mapping = {
    'heat': ('http://f5-openstack-heat.readthedocs.io/en/latest', None),
}
EOS
$ cat <<EOS >> index.rst
:ref:\`heat:Overview\`
:ref:\`heat:F5®  BIG-IP®  VE: Standalone, 3-nic\`
EOS
$ make html
...
/Users/tkomiya/work/tmp/test/index.rst:23: WARNING: undefined label: heat:f5® big-ip® ve: standalone, 3-nic (if the link has no caption the label must precede a section header)
```

Memo:
I can refer the same named section on local doc using autosectionlabel.
So I believe this problem came from the combination of inersphinx and autosectionlabel.

**Comment by jputrino** [comment]
I'm also able to refer to named sections just within one repo, so I agree that the issue is with the combination of intersphinx and autosectionlabel. 

Thanks!

**Comment by tk0miya** [comment]
Finally, I found why it happened. Current `sphinx.ext.intersphinx` has a bug which can't reference a label including both whitespaces and colons::

```
.. _foo bar:baz:

could not refer this section
=============================
```

**Comment by tk0miya** [comment]
Now I fixed this at 211fd35.
But the fix is a little ad-hoc. So please reopen again if you get another trouble.

Thank you for reporting
